### PR TITLE
Add disk space requirement in WAL doc

### PR DESCRIPTION
**Pull request by codesome**
This is will greatly in planning out the disk space requirement for WAL. The numbers would change a little with https://github.com/cortexproject/cortex/pull/2436, which I will take care in that PR.

/cc @tomwilkie

### CODE REVIEW

File: docs/production/ingesters-with-wal.md [hunk 1]
```diff
@@ -29,6 +29,34 @@ _The WAL is currently considered experimental._
 
 2. As there are no transfers between ingesters, the tokens are stored and recovered from disk between rollout/restarts. This is [not a new thing](https://github.com/cortexproject/cortex/pull/1750) but it is effective when using statefulsets.
 
+## Disk space requirement
+
+Based on tests in real worl
```
Comment (tomwilkie) [inline docs/production/ingesters-with-wal.md:34]:
```suggestion
Based on tests in real world:
```

File: docs/production/ingesters-with-wal.md [hunk 2]
```diff
@@ -29,6 +29,34 @@ _The WAL is currently considered experimental._
...
+| Total size of WAL | 4160 MiB |
+| Steady state usage | 2 x 1410 + 2 x 4160 = ~11 GiB |
+| Peak usage | 3 x 1410 + 3  x 4160 = ~17 GiB |
+
+For 1M series at 15s scrape interval with checkpoint duration of 30m
+
+|  |  |
+|:-|:-|
+| Steady state usage | 11 / 1.2 = ~9.2 GiB |
+| Peak usage | 17 / 1.2 = ~14.2 |
```
Comment (tomwilkie) [inline docs/production/ingesters-with-wal.md:56]:
```suggestion
| Peak usage | 17 GiB / 1.2 = ~14.2 GiB |
```

File: docs/production/ingesters-with-wal.md [hunk 3]
```diff
@@ -29,6 +29,34 @@ _The WAL is currently considered experimental._
...
+| Size per WAL file | 32 MiB (reduced from Prometheus) |
+| Total size of WAL | 4160 MiB |
+| Steady state usage | 2 x 1410 + 2 x 4160 = ~11 GiB |
+| Peak usage | 3 x 1410 + 3  x 4160 = ~17 GiB |
+
+For 1M series at 15s scrape interval with checkpoint duration of 30m
+
+|  |  |
+|:-|:-|
+| Steady state usage | 11 / 1.2 = ~9.2 GiB |
```
Comment (tomwilkie) [inline docs/production/ingesters-with-wal.md:55]:
```suggestion
| Steady state usage | 11 GiB / 1.2 = ~9.2 GiB |
```

File: docs/production/ingesters-with-wal.md [hunk 4]
```diff
@@ -29,6 +29,34 @@ _The WAL is currently considered experimental._
 
 2. As there are no transfers between ingesters, the tokens are stored and recovered from disk between rollout/restarts. This is [not a new thing](https://github.com/cortexproject/cortex/pull/1750) but it is effective when using statefulsets.
 
+## Disk space requirement
```
Comment (tomwilkie) [inline docs/production/ingesters-with-wal.md:32]:
```suggestion
## Disk space requirements
```

File: docs/production/ingesters-with-wal.md [hunk 5]
```diff
@@ -29,6 +29,34 @@ _The WAL is currently considered experimental._
...
+
+|  |  |
+|:-|:-|
+| Size of 1 checkpoint for 1.2M series | 1410 MiB |
+| Avg checkpoint size per series | 1.2 KiB |
+| No. of WAL files between checkpoints (30m checkpoint) | 30 mins x 87 / 20mins = 130 |
+| Size per WAL file | 32 MiB (reduced from Prometheus) |
+| Total size of WAL | 4160 MiB |
+| Steady state usage | 2 x 1410 + 2 x 4160 = ~11 GiB |
+| Peak usage | 3 x 1410 + 3  x 4160 = ~17 GiB |
```
Comment (pstibrany) [inline docs/production/ingesters-with-wal.md:49]: Units missing here. Math seems wrong: 3 * 1410 MiB + 3  * 4160 MiB = 16.32 GiB

File: docs/production/ingesters-with-wal.md [hunk 6]
```diff
@@ -29,6 +29,34 @@ _The WAL is currently considered experimental._
...
+* This peaks at 3 checkpoints and 3 lots of WAL momentarily, as we remove the old checkpoints.
+
+|  |  |
+|:-|:-|
+| Size of 1 checkpoint for 1.2M series | 1410 MiB |
+| Avg checkpoint size per series | 1.2 KiB |
+| No. of WAL files between checkpoints (30m checkpoint) | 30 mins x 87 / 20mins = 130 |
+| Size per WAL file | 32 MiB (reduced from Prometheus) |
+| Total size of WAL | 4160 MiB |
+| Steady state usage | 2 x 1410 + 2 x 4160 = ~11 GiB |
```
Comment (pstibrany) [inline docs/production/ingesters-with-wal.md:48]: Units missing here. Math checks out.

File: docs/production/ingesters-with-wal.md [hunk 7]
```diff
@@ -29,6 +29,34 @@ _The WAL is currently considered experimental._
...
+* Numbers from an ingester with 1.2M series, ~80k samples/s ingested and ~15s scrape interval.
+* Checkpoint period was 20mins, so we need to scale up the number of WAL files to account for the default of 30mins.
+* At any given point, we have 2 complete checkpoints present on the disk and a 2 sets of WAL files between checkpoints (and now).
+* This peaks at 3 checkpoints and 3 lots of WAL momentarily, as we remove the old checkpoints.
+
+|  |  |
+|:-|:-|
+| Size of 1 checkpoint for 1.2M series | 1410 MiB |
+| Avg checkpoint size per series | 1.2 KiB |
+| No. of WAL files between checkpoints (30m checkpoint) | 30 mins x 87 / 20mins = 130 |
```
Comment (pstibrany) [inline docs/production/ingesters-with-wal.md:45]: Where does 87 come from?

File: docs/production/ingesters-with-wal.md [hunk 8]
```diff
@@ -29,6 +29,34 @@ _The WAL is currently considered experimental._
...
+| Total size of WAL | 4160 MiB |
+| Steady state usage | 2 x 1410 + 2 x 4160 = ~11 GiB |
+| Peak usage | 3 x 1410 + 3  x 4160 = ~17 GiB |
+
+For 1M series at 15s scrape interval with checkpoint duration of 30m
+
+|  |  |
+|:-|:-|
+| Steady state usage | 11 GiB / 1.2 = ~9.2 GiB |
+| Peak usage | 17 GiB / 1.2 = ~14.2 GiB |
```
Comment (pstibrany) [inline docs/production/ingesters-with-wal.md:56]: What is 1.2 here?

File: docs/production/ingesters-with-wal.md (hunk 5 above)
Comment (codesome) [reply docs/production/ingesters-with-wal.md:49]: Rounding off to the next number seems safer when planning the resources, Maybe we could have 16.5, but would prefer 17. I will add the units there.

File: docs/production/ingesters-with-wal.md (hunk 7 above)
Comment (codesome) [reply docs/production/ingesters-with-wal.md:45]: As mentioned, real-world test :) 87 files for 20 mins (an upper estimate)

File: docs/production/ingesters-with-wal.md (hunk 8 above)
Comment (codesome) [reply docs/production/ingesters-with-wal.md:56]: The numbers are for 1.2M series, so dividing it by 1.2 for numbers for 1M series

File: docs/production/ingesters-with-wal.md (hunk 7 above)
Comment (codesome) [reply docs/production/ingesters-with-wal.md:45]: Added it in the description above.

File: docs/production/ingesters-with-wal.md (hunk 5 above)
Comment (pstibrany) [reply docs/production/ingesters-with-wal.md]: "Rounding" from 16.32 GiB to 17 GiB is wrong. We can write ~16.3 GiB, which would be correct, and leave rounding up for planning the resources to the reader, who will surely be smart enough to add extra capacity.

**Comment by pracucci** [comment]
@codesome The table rendering is not working fine when building the website:
Screen Shot 2020-04-22 at 17 50 43

**Comment by pracucci** [comment]
@codesome I've pushed a commit to fix the table layout for you. Unfortunately you need to also fill the table headers: feel free to rename column titles.

Screen Shot 2020-04-22 at 18 16 13

**Comment by pstibrany** [comment]
> @codesome I've pushed a commit to fix the table layout for you. Unfortunately you need to also fill the table headers: feel free to rename column titles.

Does anything need to be fixed here before merging?

**Comment by codesome** [comment]
@pracucci Thanks for adding the headers!
@pstibrany no changes from my side.

**Pull request merged**